docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+safety. Reflection can be powerful, but it can also lead to code that is hard to
+understand, hard to debug, and prone to runtime errors. By not supporting
+reflection, Gno encourages you to write code that is explicit, clear, and easy
+to understand.
+
+We're currently in the process of considering whether to add reflection support
+or not, or perhaps in a privileged mode for very rare libraries. But for now,
+when you're writing Gno code, remember: explicit is better than implicit, and
+clear code is better than clever code.
+
+## Gno good practices
+
+### Package naming and organization
+
+Your package name should match the folder name. This helps to prevent having
+named imports, which can make your code more difficult to understand and
+maintain. By matching the package name with the folder name, you can ensure that
+your imports are clear and intuitive.
+
+Ideally, package names should be short and human-readable. This makes it easier
+for other developers to understand what your package does at a glance. Avoid
+using abbreviations or acronyms unless they are widely understood.
+
+Packages and realms can be organized into subfolders. However, consider that the
+best place for your main project will likely be `r/NAMESPACE/DAPP`, similar
+to how repositories are organized on GitHub.
+
+If you have multiple sublevels of realms, remember that they are actually
+independent realms and won't share data. A good usage could be to have an
+ecosystem of realms, where one realm is about storing the state, another one
+about configuration, etc. But in general, a single realm makes sense.
+
+You can also create small realms to create your ecosystem. For example, you
+could centralize all the authentication for all your company/organization in
+`r/NAMESPACE/auth`, and then import it from all your contracts.
+
+The `p/` prefix is different. In general, you should use top-level `p/` like
+`p/NAMESPACE/DAPP` only for things you expect people to use. If your goal is
+just to have internal libraries that you created to centralize your helpers and
+don't expect that other people will use your helpers, then you should probably
+use subfolders like `p/NAMESPACE/DAPP/foo/bar/baz`.
+
+### Define types and interfaces in pure packages (p/)
+
+In Gno, it's common to create `p/NAMESPACE/DAPP` for defining types and
+interfaces, and `r/NAMESPACE/DAPP` for the runtime, especially when the goal is
+for the realm to become a standard that could be imported by `p/`.
+
+The reason for this is that `p/` can only import `p/`, while `r/` can import
+anything. This separation allows you to define standards in `p/` that can be
+used across multiple realms and packages.
+
+In general, you can just write your `r/` to be an app. But if for some reason
+you introduce a concept that should become a standard, it makes sense to have a
+dedicated `p/` so that people can start not only composing your realm from other
+realms but also for `p/`.
```

Comment:
```suggestion
In general, you can just write your `r/` to be an app. But if for some reason
you introduce a concept that can be reused, it makes sense to have a
dedicated `p/` so that people can re-use your logic without depending on
your realm's data.
```